tests/Dialect/DIP/correlation2D_invalid_type.mlir

```diff
@@ -0,0 +1,61 @@
+                                                                 [20., 21., 22.]]>
+
+
+memref.global "private" @global_identity_f32 : memref<3x3xf32> = dense<[[0., 0., 0.],
+                                                                    [0., 1., 0.],
+                                                                    [0., 0., 0.]]>
+
+memref.global "private" @global_output_f32 : memref<3x3xf32> = dense<[[0., 0., 0.],
+                                                                  [0., 0., 0.],
+                                                                  [0., 0., 0.]]>
+
+memref.global "private" @global_input_i32 : memref<3x3xi32> = dense<[[0 , 1 , 2 ],
+                                                                 [10, 11, 12],
+                                                                 [20, 21, 22]]>
+
+
+memref.global "private" @global_identity_i32 : memref<3x3xi32> = dense<[[0, 0, 0],
+                                                                    [0, 1, 0],
+                                                                    [0, 0, 0]]>
+
+memref.global "private" @global_output_i32 : memref<3x3xi32> = dense<[[0, 0, 0],
+                                                                  [0, 0, 0],
+                                                                  [0, 0, 0]]>
+
+func.func @main() -> i32 {
+  %input_f32 = memref.get_global @global_input_f32 : memref<3x3xf32>
+  %identity_f32 = memref.get_global @global_identity_f32 : memref<3x3xf32>
+  %output_f32 = memref.get_global @global_output_f32 : memref<3x3xf32>
+  %c_f32 = arith.constant 0. : f32 
+
+  %input_i32 = memref.get_global @global_input_i32 : memref<3x3xi32>
+  %identity_i32 = memref.get_global @global_identity_i32 : memref<3x3xi32>
+  %output_i32 = memref.get_global @global_output_i32 : memref<3x3xi32>
+  %c_i32 = arith.constant 0 : i32 
+
+  %x = arith.constant 1 : index
+  %y = arith.constant 1 : index
+
+  dip.corr_2d <CONSTANT_PADDING> %input_i32, %identity_f32, %output_f32, %x, %x, %c_f32 : memref<3x3xi32>, memref<3x3xf32>, memref<3x3xf32>, index, index, f32
+  // CHECK: 'dip.corr_2d' op input, kernel, output and constant must have the same element type
+
+  dip.corr_2d <CONSTANT_PADDING> %input_f32, %identity_i32, %output_f32, %x, %x, %c_f32 : memref<3x3xf32>, memref<3x3xi32>, memref<3x3xf32>, index, index, f32
+  // CHECK: 'dip.corr_2d' op input, kernel, output and constant must have the same element type
+
+  dip.corr_2d <CONSTANT_PADDING> %input_f32, %identity_f32, %output_i32, %x, %x, %c_f32 : memref<3x3xf32>, memref<3x3xf32>, memref<3x3xi32>, index, index, f32
+  // CHECK: 'dip.corr_2d' op input, kernel, output and constant must have the same element type
+
+  dip.corr_2d <CONSTANT_PADDING> %input_f32, %identity_f32, %output_f32, %x, %x, %c_i32 : memref<3x3xf32>, memref<3x3xf32>, memref<3x3xf32>, index, index, i32
+  // CHECK: 'dip.corr_2d' op input, kernel, output and constant must have the same element type
+
```

Comment:
Please add a test for [this](https://github.com/ArtemSkrebkov/buddy-mlir/blob/7d7a8db947e703fd3ae2229da3b5aa19de64f763/lib/Conversion/LowerDIP/LowerDIPPass.cpp#L109) condition as well.

Reply:
Updated `correlation2D_invalid_type.mlir` to check the condition as well.